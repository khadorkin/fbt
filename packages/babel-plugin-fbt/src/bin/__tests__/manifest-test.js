/**
 * Copyright 2004-present Facebook. All Rights Reserved.
 *
 *
 * @emails oncall+internationalization
 * @format
 */
'use strict';

jest.disableAutomock();

const childProcess = require.requireActual('child_process');
const fs = require('fs');
const path = require('path');

describe('manifest', () => {
  it('should extract strings', () => {
    const packageRoot = path.resolve(__dirname, '../../');
    const srcPath = 'bin/__fixtures__/';
    const srcManifestPath = path.join(srcPath, '.src_manifest.json');
    const enumManifestPath = path.join(srcPath, '.enum_manifest.json');

    var child = childProcess.spawnSync(
      'node',
      [
        require.resolve('../manifest'),
        '--src=' + srcPath,
        '--src-manifest=' + srcManifestPath,
        '--enum-manifest=' + enumManifestPath,
      ],
      {cwd: packageRoot},
    );

    if (
      (child.stderr && child.stderr.toString() !== '') ||
      child.error ||
      child.status !== 0
    ) {
      throw new Error(
        (child.stderr && child.stderr.toString()) ||
          child.error ||
          'Child process exited with code ' + child.status,
      );
    }

    expect(
      fs.readFileSync(path.join(packageRoot, srcManifestPath), {
        encoding: 'utf8',
      }),
    ).toMatchSnapshot();
    expect(
      fs.readFileSync(path.join(packageRoot, enumManifestPath), {
        encoding: 'utf8',
      }),
    ).toMatchSnapshot();
  });
});
