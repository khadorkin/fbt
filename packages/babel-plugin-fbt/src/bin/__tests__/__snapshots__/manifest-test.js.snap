// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manifest should extract strings 1`] = `
"{
  \\"bin/__fixtures__/.enum_manifest.json\\": []
}"
`;

exports[`manifest should extract strings 2`] = `
"{
  \\"CommonJS$FbtEnum\\": {
    \\"id1\\": \\"groups\\",
    \\"id2\\": \\"photos\\",
    \\"id3\\": \\"videos\\"
  },
  \\"ExportDefault$FbtEnum\\": {
    \\"id1\\": \\"groups\\",
    \\"id2\\": \\"photos\\",
    \\"id3\\": \\"videos\\"
  }
}"
`;
